tailwind.config.js

```diff
@@ -270,11 +273,44 @@ module.exports = {
 
   plugins: [
     function({ addVariant, e }) {
+      const SCALE_VARIANTS = [
+        { name: 'scaleXs', size: 'xs' },
+        { name: 'scaleS', size: 's' },
+        { name: 'scaleM', size: 'm' },
+        { name: 'scaleL', size: 'l' },
+        { name: 'scaleXl', size: 'xl' }
+      ]
+      const PSEUDO_SELECTORS = [ 'hover', 'focus', 'active' ]
       addVariant('rtl', ({ modifySelectors, separator }) => {
         modifySelectors(({ className }) => {
-          return `[dir="rtl"] .${e(`rtl${separator}${className}`)}`
+          return `:host([dir="rtl"]) .${e(`rtl${separator}${className}`)}`
         })
       })
+      addVariant('dark', ({ modifySelectors, separator }) => {
```

Comment:
I added variants for dark-themed components, scale variants for components, and dark-pseudoselector + rtl-scale variants as well.  I tried to get the syntax for these combination variants to reflect our normal syntax (ie. `first-variant:second-variant:class-name`), but the variants did not seem to work when two colons were included in the class name.  I spent awhile trying to play around with it, but ended up just using a `-` between the first and second variants instead.